kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/CreateExternalOrderCall.java

```diff
@@ -129,35 +160,44 @@ public void onResponse(final Order order) {
 					@Override
 					public void run() {
 						externalOrderCallbacks
-							.onOrderConfirmed(((JWTBodyPaymentConfirmationResult) order.getResult()).getJwt());
+							.onOrderConfirmed(((JWTBodyPaymentConfirmationResult) order.getResult()).getJwt(), order);
 					}
 				});
 
 			}
 
 			@Override
 			public void onFailure(final ApiException e) {
-				runOnMainThread(new Runnable() {
-					@Override
-					public void run() {
-						externalOrderCallbacks
-							.onOrderFailed(ErrorUtil.fromApiException(e));
-					}
-				});
+				onOrderFailed(ErrorUtil.fromApiException(e));
 			}
 		}).start();
 	}
 
-	private void runOnMainThread(Runnable runnable) {
-		mainThreadExecutor.execute(runnable);
-	}
+	private void onOrderFailed(final KinEcosystemException exception) {
```

Comment:
`onOrderFailed` this method being called if: `createExternalOrderSync` failed and also from `getOrder` failed.
but you are sending only `SpendOrderCreationFailed` should be `SpendOrderFailed` on `getOrder` failed method.